### PR TITLE
detect-tcp-window: Convert unittests to new FAIL/PASS API v1

### PR DESCRIPTION
**Pull request by hadiqaalamdar**
Task #6339

- [x] I have read the contributing guide lines at https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/6339) ticket:

Describe changes:
- Updated detect-tcp-window unit tests to FAIL/PASS APIs

### CODE REVIEW

File: src/detect-tcp-window.c [hunk 1]
```diff
@@ -226,87 +226,66 @@ void DetectWindowFree(DetectEngineCtx *de_ctx, void *ptr)
...
 static int DetectWindowTestParse01 (void)
 {
-    int result = 0;
     DetectWindowData *wd = NULL;
     wd = DetectWindowParse(NULL, "35402");
-    if (wd != NULL &&wd->size==35402) {
-        DetectWindowFree(NULL, wd);
-        result = 1;
-    }
+    FAIL_IF_NOT(wd != NULL &&wd->size==35402);
```
Comment (jufajardini) [inline src/detect-tcp-window.c:231]: Could we split this into two checks? 
This makes it easier to verify which check is failing if this test fails.

File: src/detect-tcp-window.c [hunk 2]
```diff
@@ -226,87 +226,66 @@ void DetectWindowFree(DetectEngineCtx *de_ctx, void *ptr)
...
-            result = 1;
-        } else {
-            printf("expected wd->negated=1 and wd->size=35402\n");
-        }
-        DetectWindowFree(NULL, wd);
-    }
-
-    return result;
+    FAIL_IF_NULL(wd);
+    FAIL_IF_NOT(wd->negated == 1 && wd->size==35402);
```
Comment (jufajardini) [inline src/detect-tcp-window.c:245]: Same request as per from the other test :)

File: src/detect-tcp-window.c [hunk 3]
```diff
@@ -327,11 +306,10 @@ static int DetectWindowTestPacket01 (void)
                               {0, 1},
                               /* packet 2 should not match */
                               {0, 0} };
-    result = UTHGenericTest(p, 3, sigs, sid, (uint32_t *) results, 2);
+    FAIL_IF_NOT(UTHGenericTest(p, 3, sigs, sid, (uint32_t *) results, 2));
```
Comment (jufajardini) [inline src/detect-tcp-window.c:309]: It is better if we are explicit here with regard to which result we expect. If result is 0, we fail the test, if it's 1, we pass, right?

**Review by jufajardini — CHANGES_REQUESTED** [review]
Can you please address the formatting error? :)

A good practice is to always run this locally before submitting a new PR, to avoid having this CI check fail.

Commit message is too long - remember we like to keep it at around ~50 characters. `detect/tcp/window: convert unittests to FAIL/PASS` should do, in this case.

File: src/detect-tcp-window.c (hunk 1 above)
Comment (hadiqaalamdar) [reply src/detect-tcp-window.c:231]: Understood, I'll make the edit

File: src/detect-tcp-window.c (hunk 3 above)
Comment (hadiqaalamdar) [reply src/detect-tcp-window.c:309]: i believe yes. I'll make the edit.

**Comment by hadiqaalamdar** [comment]
work continued on PR: https://github.com/OISF/suricata/pull/9678

**Pull request closed without merging**